### PR TITLE
.loc = to forceMove()

### PR DESCRIPTION
**Pull request by BurgerLUA**
# Overview
As requested, this PR is changed to only include all .loc = to forceMove() changes.

### CODE REVIEW

**Comment by BurgerLUA** [comment]
I had to go through every instance of drop_from_inventory to ensure that nothing goes wrong with this change. A lot of old code is present, and uses improper ways of moving/deleting stuff so I have to fix that to make this work.

**Comment by BurgerLUA** [comment]
I replaced all instances of drop_item and forcemove with drop_from_inventory. I also make it so that drop_from_inventory has an extra arg that controls whether or not the dropped item is deleted.

The reason why I'm going around and fixing everything is because most of the animations will be triggered from drop_from_inventory. It's much saner than modifying and putting animations in forceMove, as sometimes forceMove has things to move that the game shouldn't let you see.

**Comment by NonQueueingMatt** [comment]
a n i m a t e d   t h r o w i n g

Seriously though this is really sweet.

File: code/__defines/dview.dm [hunk 1]
```diff
@@ -1,14 +1,14 @@
 //DVIEW defines
 
 #define FOR_DVIEW(type, range, center, invis_flags) \
-	dview_mob.loc = center; \
+	dview_mob.forceMove(center); \
```
Comment (skull132) [inline code/__defines/dview.dm]: Is there a reason behind this change? DVIEW mob is supposed to be unaffected by as many things as possible, so bypassing the Entered hooks might actually be worth it here.

File: code/game/machinery/doors/brigdoors.dm [hunk 2]
```diff
@@ -267,11 +267,8 @@
 	if( istype( O, /obj/item/weapon/paper/incident ))
 		if( !incident )
 			if( import( O, user ))
-				usr.drop_item()
-				O.loc = src
-
 				ping( "\The [src] pings, \"Successfully imported incident report!\"" )
-				qdel( O )
+				usr.drop_from_inventory(O,get_turf(src),TRUE)
```
Comment (skull132) [inline code/game/machinery/doors/brigdoors.dm]: since we're here and this is one line. `usr` => `user`.

**Comment by BurgerLUA** [comment]
I'll be splitting it when I have time tomorrow.

File: code/__defines/dview.dm (hunk 1 above)
Comment (Lohikar) [reply code/__defines/dview.dm]: This should remain as direct loc setting, if only for performance reasons. This mob is moved around a LOT.

File: code/defines/obj.dm [hunk 3]
```diff
@@ -11,8 +11,8 @@
 		switch(alert("Travel back to ss13?",,"Yes","No"))
 			if("Yes")
 				if(user.z != src.z)	return
-				user.loc.loc.Exited(user)
-				user.loc = pick(latejoin)
+				user.loc.loc.Exited(user) //what the fuck is this
```
Comment (skull132) [inline code/defines/obj.dm]: This is something that should be no longer necessary. forceMove ensures in this being proper.

File: code/game/machinery/doors/brigdoors.dm [hunk 4]
```diff
@@ -263,15 +263,12 @@
...
+/obj/machinery/door_timer/attackby(obj/item/O as obj, var/mob/user as mob)
 	if( istype( O, /obj/item/weapon/paper/incident ))
 		if( !incident )
 			if( import( O, user ))
-				usr.drop_item()
-				O.loc = src
-
 				ping( "\The [src] pings, \"Successfully imported incident report!\"" )
-				qdel( O )
+				user.drop_from_inventory(O,get_turf(src),TRUE)
```
Comment (skull132) [inline code/game/machinery/doors/brigdoors.dm]: Drop from inventory != qdel wtf bbq broskidoodle.

File: code/game/machinery/washing_machine.dm [hunk 5]
```diff
@@ -79,9 +79,8 @@
 	if(istype(W,/obj/item/weapon/pen/crayon) || istype(W,/obj/item/weapon/stamp))
 		if( state in list(	1, 3, 6 ) )
 			if(!crayon)
-				user.drop_item()
+				user.drop_from_inventory(W,src)
 				crayon = W
```
Comment (skull132) [inline code/game/machinery/washing_machine.dm]: Erm, this isn't correct. This should be `crayon.forceMove(src)`

File: code/game/machinery/washing_machine.dm [hunk 6]
```diff
@@ -144,8 +143,7 @@
 
 		if(contents.len < 5)
 			if ( state in list(1, 3) )
-				user.drop_item()
-				W.loc = src
```
Comment (skull132) [inline code/game/machinery/washing_machine.dm]: `W.forceMove(src)`

File: code/game/machinery/doors/brigdoors.dm (hunk 4 above)
Comment (BurgerLUA) [reply code/game/machinery/doors/brigdoors.dm]: Oh, I neglected to mention that drop_from_inventory also now has a (qdel) proc built in if the third argument is set to TRUE.

File: code/game/machinery/washing_machine.dm (hunk 5 above)
Comment (BurgerLUA) [reply code/game/machinery/washing_machine.dm]: user.drop_from_inventory(item,target) acts as a item.forceMove(target)

File: code/game/antagonist/antagonist_equip.dm [hunk 7]
```diff
@@ -6,7 +6,7 @@
 	// This could use work.
 	if(flags & ANTAG_CLEAR_EQUIPMENT)
 		for(var/obj/item/thing in player.contents)
-			player.drop_from_inventory(thing)
+			player.drop_from_inventory(thing,enable_animations = FALSE)
```
Comment (Lohikar) [inline code/game/antagonist/antagonist_equip.dm]: `enable_animations`?

**Comment by BurgerLUA** [comment]
!review

File: code/game/antagonist/antagonist_equip.dm (hunk 7 above)
Comment (BurgerLUA) [reply code/game/antagonist/antagonist_equip.dm]: Leftover code from when this PR used to have the animation element put in.

File: code/defines/obj.dm (hunk 3 above)
Comment (BurgerLUA) [reply code/defines/obj.dm]: So how should this look? just user.ForceMove(loc) ?

File: code/game/machinery/doors/brigdoors.dm (hunk 4 above)
Comment (Lohikar) [reply code/game/machinery/doors/brigdoors.dm]: ...why not just qdel() the object and let Destroy() call DFI.

File: code/game/antagonist/antagonist_equip.dm (hunk 7 above)
Comment (Lohikar) [reply code/game/antagonist/antagonist_equip.dm]: It'll have to be removed then as it'll cause runtimes.

File: code/game/machinery/doors/brigdoors.dm (hunk 4 above)
Comment (BurgerLUA) [reply code/game/machinery/doors/brigdoors.dm]: It's faster to check a bool than it is to check whether or not the object is being held by a person and to call a drop_from_inventory.

Comment (Lohikar) [reply code/game/machinery/doors/brigdoors.dm]: This isn't performance critical code.

File: code/defines/obj.dm (hunk 3 above)
Comment (skull132) [reply code/defines/obj.dm]: Yes.

File: code/game/antagonist/antagonist_equip.dm (hunk 7 above)
Comment (skull132) [reply code/game/antagonist/antagonist_equip.dm]: Lohikar's commentis correct. Dew it.

**Comment by Lohikar** [comment]
I don't think a qdel argument on DFI is a good idea either, in those cases the item should just be qdel()'d and the Destroy() stack allowed to clean it up.

**Comment by skull132** [comment]
@BurgerLUA boop.

**Comment by BurgerLUA** [comment]
@skull132 Move this to the august update. There is a lot of things I need to test, still, and with the requested changes it's going to take more than one sitdown to make everything up to par.

**Comment by BurgerLUA** [comment]
do not merge this you mad man smh
I could make a change and it would still say you approve

**Comment by BurgerLUA** [comment]
I am going to make it so "heil hitler" is displayed every round start and then screenshot skull approving this.

**Pull request merged**